nss-day-cohort-60/national-parks-django-api
# Blog_View implemented

**Notes**
Adds functionality for clients to do the following:
- send a GET request for all blogs
- send a GET request for a singular blog
- send a GET request for all blogs related to a single park
- send a GET request for all blogs related to a single user
- send a POST request for a new blog
- send a PUT request to edit any blog
- send a DELETE request for any blog

**Steps to Test**
test GET request for all blogs:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Send a GET request to http://localhost:8000/blogs
6. You should see 8 blogs, with park and user info embedded in the response

test GET request for a singular blog:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Send a GET request to http://localhost:8000/blogs/4
6. You should see a blog titled "Preserve the Smokies!" in the response

test GET request for all blogs related to a certain park:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Send a GET request to http://localhost:8000/blogs?park_id=1
6. You should only see blogs related to Joshua Tree national park in the response

test GET request for all blogs related to a certain user:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Send a GET request to http://localhost:8000/blogs?user_id=1
6. You should only see blogs related to Legolas Greenleaf in the response

test POST request for a new blog:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Fill out the body of a POST request to http://localhost:8000/blogs
6. It has to have title, post_body, date_created, park, and photo
7. I struggled filling out date_created so here's a generic one if you need it: `2011-10-01 15:26`
8. Send the request
9. You should only see a JSON representation of the blog you just posted in the response

test PUT request for a blog:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Send a PUT request to http://localhost:8000/blogs/9 (should be the blog you just posted)
7. Your request should only have title, post_body, park, and photo in the request body
8. Once sent, you should see a 204 response code and nothing else in the response
9. Send a GET request to http://localhost:8000/blogs/9 and make sure your changes appear

test DELETE request for a blog:
1. `git fetch hp-blog-views`
2. `git checkout hp-blog-views`
3. `manage.py runserver`
4. Open postman
5. Send a DELETE request to http://localhost:8000/blogs/9 (should be the blog you just edited)
6. Send a GET request to http://localhost:8000/blogs and make sure it's not there any more.

Simple enough, right team?
